tutor/src/components/scroll-tracker.js

```diff
@@ -1,127 +1,20 @@
+  useEffect(() => {
+    const _onScroll = () => {
+      onScroll({
+        scrollTop: ScrollElement.scrollTop,
+        clientHeight: ScrollElement.clientHeight,
+      });
     };
-  },
-
-  setScrollTopBuffer() {
-    const scrollTopBuffer = GetPositionMixin.getTopPosition(ReactDOM.findDOMNode(this));
-    return this.setState({ scrollTopBuffer });
-  },
-
-  setScrollPoint(scrollPoint, scrollState) {
-    const scrollPointData = _.extend({ scrollPoint }, scrollState);
-    this.state.scrollPoints.push(scrollPointData);
-    return this.sortScrollPoints();
-  },
-
-  unsetScrollPoint(unsetScrollPoint) {
-    this.state.scrollPoints = _.reject(this.state.scrollPoints, scrollPoint => scrollPoint.scrollPoint === unsetScrollPoint);
-    return this.sortScrollPoints();
-  },
 
-  sortScrollPoints() {
-    const sortedDescScrollPoints = _.sortBy(this.state.scrollPoints, scrollData => -1 * scrollData.scrollPoint);
-
-    return this.setState({ scrollPoints: sortedDescScrollPoints });
-  },
-
-  getScrollStateByScroll(scrollTop) {
-    const scrollState = _.find(this.state.scrollPoints, scrollData => {
-      return scrollTop > (scrollData.scrollPoint - this.state.scrollTopBuffer - 2);
-    });
-
-    return scrollState || _.last(this.state.scrollPoints);
-  },
-
-  getScrollStateByKey(stepKey) {
-    const scrollStateIndex = _.findLastIndex(this.state.scrollPoints, scrollData => scrollData.key === stepKey);
-
-    return this.state.scrollPoints[scrollStateIndex];
-  },
-
-  setScrollState() {
-    const scrollState = this.getScrollStateByScroll(this.state.scrollTop);
-    this.setState({ scrollState });
-
-    return this.props.setScrollState(scrollState);
-  },
-
-  componentDidMount() {
-    this.setScrollTopBuffer();
-    if (this.props.currentStep != null) { this.scrollToKey(this.props.currentStep); }
-  },
-
-  UNSAFE_componentWillUpdate(nextProps, nextState) {
-    const willScrollStateKeyChange = !(nextState.scrollState.key === this.state.scrollState.key);
-    if (willScrollStateKeyChange) { this.props.goToStep(nextState.scrollState.key); }
-  },
-
-  componentDidUpdate(prevProps, prevState) {
-    const doesScrollStateMatch = (prevState.scrollState.key === __guard__(this.getScrollStateByScroll(this.state.scrollTop), x => x.key));
-    const didCurrentStepChange = !(this.props.currentStep === (prevState.scrollState != null ? prevState.scrollState.key : undefined));
-
-    if (!doesScrollStateMatch) {
-      this.setScrollState();
-    } else if (didCurrentStepChange) {
-      this.scrollToKey(this.props.currentStep);
-    }
-  },
-
-  scrollToKey(stepKey) {
-    const scrollState = this.getScrollStateByKey(stepKey);
-    return window.scrollTo(0, ((scrollState != null ? scrollState.scrollPoint : undefined) - this.state.scrollTopBuffer));
-  },
+    window.addEventListener('scroll', debounce(_onScroll, 10), { passive: true });
+    return () => window.removeEventListener('scroll', _onScroll);
+  });
```

Comment:
You'll need to use the exact same function on both addEvent and removeEvent, otherwise removeEvent won't work.

To do so you can move the debounce up like so: `const _onScroll => debounce( () => { ... } )`